src/partials/welcome.ts

```diff
@@ -267,6 +267,24 @@ class WelcomeController {
          return this.inMemorySession.getPassword()
     }
 
+    /**
+     * Return the OS name of the last connected device, if known.
+     */
+    public get lastConnectedClientOS(): threema.OperatingSystem | undefined {
+        try {
+            const lastConnectedClientOS = localStorage.getItem("lastConnectedClientOS");
+            if (lastConnectedClientOS === "ios") {
+                return threema.OperatingSystem.Ios
+            } else if (lastConnectedClientOS === "android") {
+                return threema.OperatingSystem.Android
+            } else {
+                this.log.error("Unexpected value for lastConnectedClientOS was returned from localStorage");
+            }
```

Comment:
If the local storage key is missing or empty, currently many errors are logged. In that case, I'd just return `undefined` without logging an error.

Reply:
Amended!